karapace/kafka_rest_apis/__init__.py

```diff
@@ -134,37 +146,214 @@ def _add_kafka_rest_routes(self) -> None:
+
+    async def seek_end_consumer_offsets(
+        self, group_name: str, instance: str, content_type: str, *, request: HTTPRequest
+    ) -> None:
+        proxy = self.get_user_proxy(request)
+        await proxy.seek_end_consumer_offsets(group_name, instance, content_type, request=request)
+
+    async def update_consumer_offsets(
+        self, group_name: str, instance: str, content_type: str, *, request: HTTPRequest
+    ) -> None:
+        proxy = self.get_user_proxy(request)
+        await proxy.update_consumer_offsets(group_name, instance, content_type, request=request)
+
+    async def fetch(self, group_name: str, instance: str, content_type: str, *, request: HTTPRequest) -> None:
+        proxy = self.get_user_proxy(request)
+        await proxy.fetch(group_name, instance, content_type, request=request)
+
+    async def create_consumer(self, group_name: str, content_type: str, *, request: HTTPRequest) -> None:
+        proxy = self.get_user_proxy(request)
+        await proxy.create_consumer(group_name, content_type, request=request)
+
+    async def delete_consumer(self, group_name: str, instance: str, content_type: str, *, request: HTTPRequest) -> None:
+        proxy = self.get_user_proxy(request)
+        await proxy.delete_consumer(group_name, instance, content_type)
+
+    async def partition_offsets(self, content_type: str, *, topic: str, partition_id: str, request: HTTPRequest):
+        proxy = self.get_user_proxy(request)
+        await proxy.partition_offsets(content_type, topic=topic, partition_id=partition_id)
+
+    async def partition_details(self, content_type: str, *, topic: str, partition_id: str, request: HTTPRequest):
+        proxy = self.get_user_proxy(request)
+        await proxy.partition_details(content_type, topic=topic, partition_id=partition_id)
+
+    async def partition_publish(self, topic: str, partition_id: str, content_type: str, *, request: HTTPRequest) -> None:
+        proxy = self.get_user_proxy(request)
+        await proxy.partition_publish(topic, partition_id, content_type, request=request)
+
+    async def list_partitions(self, content_type: str, *, topic: str, request: HTTPRequest):
+        proxy = self.get_user_proxy(request)
+        await proxy.list_partitions(content_type, topic=topic)
+
+    async def list_topics(self, content_type: str, *, request: HTTPRequest):
+        proxy = self.get_user_proxy(request)
+        await proxy.list_topics(content_type)
+
+    async def topic_details(self, content_type: str, *, topic: str, request: HTTPRequest):
+        proxy = self.get_user_proxy(request)
+        await proxy.topic_details(content_type, topic=topic)
+
+    async def topic_publish(self, topic: str, content_type: str, *, request: HTTPRequest) -> None:
+        proxy = self.get_user_proxy(request)
+        await proxy.topic_publish(topic, content_type, request=request)
+
+
+class UserRestProxy:
```

Comment:
Could this track if there is auth error? I am thinking that the proxy could use backoff so it would not try constantly?